components/vision.py

```diff
@@ -10,6 +10,6 @@ def __init__(self) -> None:
 
     def get_vision_data(self) -> Tuple[float, float, float]:
```

Comment:
This type hint is now wrong...

Reply:
Just pointing out that something is wrong is not the most helpful review comment. Suggest a correct way.

In this case, a better type hint is to use a union type: 
def get_vision_data(self) -> Tuple[Union[float, None], Union[float, None], Union[float, None]]
but Python has syntactic sugar for Union[x, None], so this can be reduced to
def get_vision_data(self) -> Tuple[Optional[float], Optional[float], Optional[float]]

However, returning a tuple of 3 Nones seems awkward. Better yet would be to return either a tuple of floats or a single None:
def get_vision_data(self) -> Optional[Tuple[]float, float, float]]